--- js/fullcalendar.js.orig	2011-06-04 13:45:44.000000000 -0600
+++ js/fullcalendar.js	2011-06-05 18:58:59.000000000 -0600
@@ -500,8 +500,8 @@
 	}
 	
 	
-	function refetchEvents() {
-		fetchEvents(currentView.visStart, currentView.visEnd); // will call reportEvents
+	function refetchEvents(source) {
+		fetchEvents(currentView.visStart, currentView.visEnd, source); // will call reportEvents
 	}
 	
 	
@@ -897,15 +897,16 @@
 	}
 	
 	
-	function fetchEvents(start, end) {
+	function fetchEvents(start, end, src) {
 		rangeStart = start;
 		rangeEnd = end;
 		cache = [];
 		var fetchID = ++currentFetchID;
 		var len = sources.length;
-		pendingSourceCnt = len;
+		pendingSourceCnt = typeof src == 'undefined' ? len : 1;
 		for (var i=0; i<len; i++) {
-			fetchEventSource(sources[i], fetchID);
+			if (typeof src == 'undefined' || src == sources[i])
+				fetchEventSource(sources[i], fetchID);
 		}
 	}
 	
